bin/host/src/interop/orchestrator.rs

```diff
@@ -0,0 +1,105 @@
+//! [SingleChainHostCli]'s [HostOrchestrator] + [DetachedHostOrchestrator] implementations.
+
+use super::{InteropFetcher, InteropHostCli, LocalKeyValueStore};
+use crate::eth::{http_provider, OnlineBlobProvider};
+use alloy_provider::{Provider, ReqwestProvider};
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use kona_host::{
+    DetachedHostOrchestrator, DiskKeyValueStore, Fetcher, HostOrchestrator, MemoryKeyValueStore,
+    SharedKeyValueStore, SplitKeyValueStore,
+};
+use kona_preimage::{HintWriter, NativeChannel, OracleReader};
+use std::{collections::HashMap, sync::Arc};
+use tokio::sync::RwLock;
+
+/// The providers required for the single chain host.
+#[derive(Debug)]
+pub struct InteropProviders {
+    /// The L1 EL provider.
+    l1_provider: ReqwestProvider,
+    /// The L1 beacon node provider.
+    blob_provider: OnlineBlobProvider,
+    /// The L2 EL providers, keyed by chain ID.
+    l2_providers: HashMap<u64, ReqwestProvider>,
+}
+
+#[async_trait]
+impl HostOrchestrator for InteropHostCli {
+    type Providers = InteropProviders;
+
+    async fn create_providers(&self) -> Result<Option<Self::Providers>> {
+        if self.is_offline() {
+            return Ok(None);
+        }
+
+        let l1_provider =
+            http_provider(self.l1_node_address.as_ref().ok_or(anyhow!("Provider must be set"))?);
```

Comment:
lets define a proper error enum for the errors thrown in this impl, and deriving `thiserror::Error` for it, has no_std support https://github.com/dtolnay/thiserror/blob/0f532e326e9a4cc6c6e30ee19ab00cb9eeb44362/Cargo.toml#L17-L28

Reply:
Glad to do - though the entire host currently uses `anyhow`. I'll create a ticket for nuking `anyhow` from the codebase, the host is the last step.

Reply:
https://github.com/op-rs/kona/issues/916